### PR TITLE
Add excel users

### PR DESCRIPTION
**Pull request by ioangrozea**

### CODE REVIEW

File: src/app/shared/components/file-uploader/file-uploader.component.ts [hunk 1]
```diff
@@ -11,23 +11,35 @@ import * as fromStore from '@app/core/store';
...
+  isInProgress$: Observable<boolean> = this.store$.pipe(
+    select(fromStore.selectUploadFileInProgress)
+  );
+  isReady$: Observable<boolean> = this.store$.pipe(
+    select(fromStore.selectUploadFileReady)
+  );
+  hasFailed$: Observable<boolean> = this.store$.pipe(
+    select(fromStore.selectUploadFileFailed)
+  );
+
```
Comment (bogdanconstantinescu) [inline src/app/shared/components/file-uploader/file-uploader.component.ts]: Oh yeah! Way more readable... ¯\_(ツ)_/¯

File: src/app/users/components/index.ts [hunk 2]
```diff
@@ -7,4 +7,3 @@ export const components: any[] = [
   UsersHeaderComponent,
   UserFormComponent
 ];
-
```
Comment (bogdanconstantinescu) [inline src/app/users/components/index.ts]: Why??? ¯\_(ツ)_/¯

File: src/app/users/containers/add-single-user/add-single-user.component.ts [hunk 3]
```diff
@@ -14,12 +14,15 @@ import { Error } from '@app/core/models/error.model';
 })
 export class AddSingleUserComponent {
   public user: User = new User();
-  private error$: Observable<Error> = this.store.select(fromStore.getUsersErrors);
+  private error$: Observable<Error> = this.store.select(
+    fromStore.getUsersErrors
```
Comment (bogdanconstantinescu) [inline src/app/users/containers/add-single-user/add-single-user.component.ts]: ¯\_(ツ)_/¯

At least do the right thing and do a `this.store.PIPE(select(fromStore.getUsersErrors))`

File: src/app/users/containers/users/users.component.ts [hunk 4]
```diff
@@ -9,17 +9,23 @@ import { Router } from '@angular/router';
...
+  );
+  public usersPageData$: Observable<PageData> = this.store.pipe(
+    select(fromStore.getUsersPageData)
+  );
+  private usersLoaded$: Observable<boolean> = this.store.pipe(
+    select(fromStore.getUsersLoaded)
+  );
+  private filter$: Observable<Filter> = this.store.pipe(
+    select(fromStore.getUsersFilter)
+  );
```
Comment (bogdanconstantinescu) [inline src/app/users/containers/users/users.component.ts]: ¯\_(ツ)_/¯

File: src/app/users/guards/users.guard.ts [hunk 5]
```diff
@@ -8,27 +8,24 @@ import * as fromStore from '../../core/store';
...
-        take(1)
-      );
+    return this.store.pipe(select(fromStore.getUsersShouldReload)).pipe(
+      tap(shouldReload => {
+        if (shouldReload) {
+          this.store.dispatch(new fromStore.LoadUsers());
+        }
+      }),
+      filter(loaded => loaded),
+      take(1)
```
Comment (bogdanconstantinescu) [inline src/app/users/guards/users.guard.ts]: ¯\_(ツ)_/¯

File: src/app/users/users.module.ts [hunk 6]
```diff
@@ -9,18 +9,8 @@ import * as fromComponents from './components';
...
-  providers: [
-    ...fromGuards.guards,
-  ]
+  imports: [CommonModule, SharedModule, UsersRoutingModule],
+  declarations: [...fromContainers.components, ...fromComponents.components],
+  providers: [...fromGuards.guards]
 })
-export class UsersModule {
-}
+export class UsersModule {}
```
Comment (bogdanconstantinescu) [inline src/app/users/users.module.ts]: ¯\_(ツ)_/¯

**Comment by RaduCStefanescu** [comment]
Hi!
@ioangrozea any chance you might be able to attend to the comments in this PR and resolve the conflicts, so we can merge this one?
Thanks!

**Comment by georgebejan** [comment]
This can be closed in favor of #59 (rebased and fixed formatting)

**Comment by RaduCStefanescu** [comment]
Closing this on as it is replaced by #59

**Pull request closed without merging**